File: 75. Sort Colors.go
func sortColors(nums []int)  {
    left, right, i := 0, len(nums)-1, 0
    for i <= right {
        if nums[i] == 0 {
            nums[i], nums[left] = nums[left], nums[i]
            i += 1
            left += 1
        } else if nums[i] == 2 {
            nums[i], nums[right] = nums[right], nums[i]
            right -= 1
        } else {
            i += 1
        }
    }
}

File: java/75. Sort Colors.java
class Solution {
    public void sortColors(int[] nums) {
        int left = 0, right = nums.length-1, i = 0;
        while (i <= right) {
            if (nums[i] == 2) {
                int temp = nums[i];
                nums[i] = nums[right];
                nums[right] = temp;
                right--;
            } else if (nums[i] == 0) {
                int temp = nums[i];
                nums[i] = nums[left];
                nums[left] = temp;
                i++;
                left++;
            } else {
                i++;
            }
        }
    }
}
